frontend/test/metabase/scenarios/visualizations/object_detail.cy.spec.js

```diff
@@ -8,30 +8,59 @@ import {
 
 import { SAMPLE_DATABASE } from "__support__/e2e/cypress_sample_database";
 
-const { ORDERS, ORDERS_ID, PRODUCTS } = SAMPLE_DATABASE;
+const { ORDERS, ORDERS_ID, PRODUCTS, PRODUCTS_ID } = SAMPLE_DATABASE;
+
+const FIRST_ORDER_ID = 9676;
+const SECOND_ORDER_ID = 10874;
+const THIRD_ORDER_ID = 11246;
+
+const TEST_QUESTION = {
+  query: {
+    "source-table": ORDERS_ID,
+    filter: [
+      "and",
+      [">", ["field", ORDERS.TOTAL, null], 149],
+      [">", ["field", ORDERS.TAX, null], 10],
+      ["not-null", ["field", ORDERS.DISCOUNT, null]],
+    ],
+  },
+};
 
 describe("scenarios > question > object details", () => {
-  const FIRST_ORDER_ID = 9676;
-  const SECOND_ORDER_ID = 10874;
-  const THIRD_ORDER_ID = 11246;
-
-  const TEST_QUESTION = {
-    query: {
-      "source-table": ORDERS_ID,
-      filter: [
-        "and",
-        [">", ["field", ORDERS.TOTAL, null], 149],
-        [">", ["field", ORDERS.TAX, null], 10],
-        ["not-null", ["field", ORDERS.DISCOUNT, null]],
-      ],
-    },
-  };
-
   beforeEach(() => {
     restore();
     cy.signInAsAdmin();
   });
 
+  it("shows correct object detail card for questions with joins (metabase #27094)", () => {
+    const questionDetails = {
+      name: "14775",
+      query: {
+        "source-table": ORDERS_ID,
+        joins: [
+          {
+            fields: "all",
+            "source-table": PRODUCTS_ID,
+            condition: [
+              "=",
+              ["field-id", ORDERS.PRODUCT_ID],
+              ["joined-field", "Products", ["field-id", PRODUCTS.ID]],
+            ],
+            alias: "Products",
+          },
+        ],
+      },
+    };
+
+    cy.createQuestion(questionDetails, { visitQuestion: true });
+
+    drillPK({ id: 1 });
+
+    cy.findByTestId("object-detail").within(() => {
+      cy.get("h2").should("contain", "Order").should("contain", 1);
```

Comment:
@alxnddr I didn't see this on time, so forgive me for the late reply.

Please do not use selectors like `h2`. That's very brittle.
You can use `cy.findByRole("heading")` instead.

Reply:
Also, wasn't this supposed to be reproduced with uuid?
I've tried to reproduce it with the Sample Database originally but wasn't able to.

Reply:
@nemanjaglumac the issue is not related to the column type